launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -190,6 +190,10 @@
       firstNonEmptyValue(SparkLauncher.DRIVER_EXTRA_CLASSPATH, conf, props) : null;
 
     List<String> cmd = buildJavaCommand(extraClassPath);
+    // Take Thrift Server as daemon
+    if (isThriftServer(mainClass)) {
```

Comment:
The alternative would be to make this `HiveThriftServer2` go through `spark-class` instead of `spark-submit`, but I think that would be a bigger change. While I'm not in love of the idea of peppering special cases inside  `SparkSubmitCommandBuilder`, one doesn't sound like a terrible idea. If there are more in the future we can look at a better solution.